packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/formdescriptioneditors/FormDescriptionEditorLifecycleControllerTests.java

```diff
@@ -0,0 +1,187 @@
+
+        Consumer<FormDescriptionEditorRefreshedEventPayload> initialFormDescriptionEditorContentConsumer = payload -> Optional.of(payload)
+                .map(FormDescriptionEditorRefreshedEventPayload::formDescriptionEditor)
+                .ifPresentOrElse(formDescriptionEditor -> {
+                    formDescriptionEditorId.set(formDescriptionEditor.getId());
+                    var firstGroup = formDescriptionEditor.getPages().get(0).getGroups().get(0);
+                    assertThat(firstGroup.getWidgets()).hasSize(3);
+
+                    var representationId = new UUIDParser().parse(formDescriptionEditor.getId()).orElseThrow(() -> new IllegalArgumentException("Invalid identifier"));
+                    this.representationDataRepository.findById(representationId)
+                            .ifPresentOrElse(representationData::set, () -> fail("Missing representation data"));
+                }, () -> fail("Missing form description editor"));
+
+        Runnable addWidget = () -> {
+            var addWidgetInput = new AddWidgetInput(
+                    UUID.randomUUID(),
+                    StudioIdentifiers.SAMPLE_STUDIO_PROJECT.toString(),
+                    formDescriptionEditorId.get(),
+                    StudioIdentifiers.GROUP_OBJECT.toString(),
+                    "Textfield",
+                    0
+            );
+            var result = this.addWidgetMutationRunner.run(addWidgetInput);
+            String typename = JsonPath.read(result, "$.data.addWidget.__typename");
+            assertThat(typename).isEqualTo(SuccessPayload.class.getSimpleName());
+        };
+
+        Consumer<FormDescriptionEditorRefreshedEventPayload> addedWidgetConsumer = payload -> Optional.of(payload)
+                .map(FormDescriptionEditorRefreshedEventPayload::formDescriptionEditor)
+                .ifPresentOrElse(formDescriptionEditor -> {
+                    var firstGroup = formDescriptionEditor.getPages().get(0).getGroups().get(0);
+                    assertThat(firstGroup.getWidgets()).hasSize(4);
+                    newWidgetId.set(firstGroup.getWidgets().get(3).getId());
+                }, () -> fail("Missing form description editor"));
+
+        Runnable reloadFormDescriptionEditor = () -> {
+            // Restore the previous state of the FormDescription
+            var deleteWidgetInput = new DeleteWidgetInput(
+                    UUID.randomUUID(),
+                    StudioIdentifiers.SAMPLE_STUDIO_PROJECT.toString(),
+                    formDescriptionEditorId.get(),
+                    newWidgetId.get());
+            var result = this.deleteWidgetMutationRunner.run(deleteWidgetInput);
+            String typename = JsonPath.read(result, "$.data.deleteWidget.__typename");
+            assertThat(typename).isEqualTo(SuccessPayload.class.getSimpleName());
+
+            TestTransaction.flagForCommit();
+            TestTransaction.end();
+            TestTransaction.start();
```

Comment:
This will trigger a refresh of the form description editor by itself, wouldn't that make the test "work" even without the reload after that? Unless I missed something, written this way the test does not prove that the feature works.

Reply:
The test should probably change to show the feature working